### ISSUE TITLE
aws_codepipeline_webhook doesn't support update

### ISSUE THREAD

**Issue by zentavr** [terraform-providers/terraform-provider-aws#8017]
### Terraform Version
```
$ terraform -v
Terraform v0.11.11
+ provider.aws v2.2.0
+ provider.github v1.3.0
+ provider.local v1.1.0
+ provider.null v2.1.0
+ provider.random v2.0.0
+ provider.template v1.0.0

Your version of Terraform is out of date! The latest version
is 0.11.13. You can update by downloading from www.terraform.io/downloads.html
```

### Affected Resource(s)
* aws_codepipeline_webhook

### Terraform Configuration Files

```hcl
# Create a Hook URL in Codepipeline
resource "aws_codepipeline_webhook" "github" {
  name  = "${var.app_environment}-${var.service_name}-webhook"
  count = "${var.enable_hooks}"

  # We are fetching the stage name dynamically
  #   The name of the action in a pipeline you want to connect to the webhook.
  #   The action must be from the source (first) stage of the pipeline.
  target_action = "${aws_codepipeline.pipeline.stage.0.action.0.name}"

  target_pipeline = "${aws_codepipeline.pipeline.name}"

  authentication = "GITHUB_HMAC"

  authentication_configuration {
    secret_token = "${sha1("${lookup(var.github, "branch")}+${lookup(var.github, "token")}")}"
  }

  filter {
    json_path    = "$.ref"
    match_equals = "refs/heads/{Branch}"
  }
}

# Wire the CodePipeline webhook into a GitHub repository.
resource "github_repository_webhook" "dashboard" {
  count      = "${var.enable_hooks}"
  repository = "${var.github["repo"]}"
  name       = "web"

  configuration {
    url          = "${aws_codepipeline_webhook.github.0.url}"
    content_type = "json"
    insecure_ssl = false
    secret       = "${sha1("${lookup(var.github, "branch")}+${lookup(var.github, "token")}")}"
  }

  events = [
    "push",
  ]
}
```

### Debug Output
```
...
...
Do you want to perform these actions in workspace "nirvana"?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes
...
...
module.codepipeline_dashboard.aws_codepipeline_webhook.github: Modifying... (ID: arn:aws:codepipeline:us-west-2:203867187697:webhook:nirvana-dashboard-webhook)
  authentication_configuration.0.secret_token: "<sensitive>" => "<sensitive>"
2019/03/20 01:18:43 [ERROR] root.codepipeline_dashboard: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

* aws_codepipeline_webhook.github: doesn't support update
2019/03/20 01:18:43 [ERROR] root.codepipeline_dashboard: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* aws_codepipeline_webhook.github: doesn't support update

```

### Panic Output

```
Error: Error applying plan:

1 error(s) occurred:
* module.codepipeline_dashboard.aws_codepipeline_webhook.github: 1 error(s) occurred:
* aws_codepipeline_webhook.github: doesn't support update
```

### Expected Behavior
A webhook should be updated or re-created (deleted/created again). If this is not possible - update the doc and tell the audience that the secret is not changable.

### Actual Behavior
Webhook cannot be updated

### Steps to Reproduce

1. Set up the secret
2. `terraform apply`
3. Change the secret
4. ...have an error ...

**Comment by zentavr** [comment]
The workaround is the next:
1. List the webhooks:
   `aws codepipeline list-webhooks --region "us-west-2"`
2. Find yours and delete that. Be very careful!!! 
   `aws codepipeline delete-webhook --region "us-west-2" --name "nirvana-dashboard-webhook"`
3. Refresh the state file with `terraform refresh`. More verbose command you can find [here](https://www.terraform.io/docs/commands/refresh.html)
4. Try to re-apply.

**Comment by russmac** [comment]
Also exists in in 0.12.5

**Comment by ewbankkit** [comment]
PR https://github.com/terraform-providers/terraform-provider-aws/pull/11387 submitted to fix this problem.

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.45.0 of the Terraform AWS Provider, Thursday next week. Thanks to @ewbankkit for the implementation. 👍
